src/constants/contracts.ts

```diff
@@ -69,3 +69,63 @@ const local: Contracts = {
 
 export const contracts: Contracts =
   NETWORK === "LOCAL" ? local : NETWORK === "TESTNET" ? testnet : mainnet;
+
+export const polygonContracts = {
```

Comment:
@SovereignAndrey @ap-justin added this here to avoid having to hardcode polygon contracts in particular files all the time, but this is still a WIP, or described through a clever use of a famous meme:
![image](https://user-images.githubusercontent.com/19427053/229084208-d0f3bd69-dece-44f6-baa5-d793d4ecd4c0.png)

### An idea on how to avoid having to copy/paste contract addresses as were doing with cosmwasm contracts

We can use the fact that TypeScript allows for importing JSON files and that it automatically adds types to the imported JSON objects ([VS Code supports types JSON objects](https://code.visualstudio.com/docs/languages/json), but even if you're using some other editor, TS supports typing JSON out-of-the-box as well, see [official docs](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-9.html#new---resolvejsonmodule)). This means that instead of copy-pasting particular contract addresses into the `contracts.ts` file, we can copy-paste the **entire file containing the contract addresses**.
Our code structure would look something like:
```
src
- constants
  - contracts-local.json
  - contracts-testnet.json
  - contracts-mainnet.json
  - contracts.ts // --> the file we currently use for contract addresses
```
And in `contracts.ts` we'd have:
```ts
import local from "./contracts-local.json";
import testnet from "./contracts-testnet.json";
import mainnet from "./contracts-mainnet.json";
import { NETWORK } from "./env";
...
export const contracts: Contracts =
  NETWORK === "LOCAL" ? local : NETWORK === "TESTNET" ? testnet : mainnet;
```

Files `contracts-local.json`, `contracts-testnet.json` and `contracts-mainnet.json` contents would just be copy/pasted contents of the current `contract-address.json` in EVM project. We could update the EVM deploy scripts to create separate `contract-address.json` files for mainnet, testnet and local and copy those in respective files in the webapp.

Additionally, if we were to get real fancy, we could create a script or update an existing one to automatically copy/paste these `contracts-[X].json` files into the webapp and auto-create a PR to desired webapp branch (using [GitHub CLI](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request?tool=cli)). This way we would never have to manually copy/paste contracts again, they would always be up-to-date on both EVM project and webapp project.

Reply:
Here's an example script that can be used to update contracts in the webapp:
https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2027/files

I tested  this during the weekend on one of my dummy GH projects and it worked flawlessly.

The script is assumed to be a part of the EVM project (created here just for easier showcasing), it can be used as standalone (run as some `yarn` script) or as part of some other script and is of course subject to change as we see fit.

Reply:
@ap-justin can you expand upon the benefit of this approach?
```ts
  export type Contract =
  | "registrar"
  | "index-fund"
  | "multisig/ap"
  | "multisig/review"
  | "accounts"
  | "halo"
  | "gift-card"
  | "gov"
  | "airdrop";
  
  type Contracts = { [key in Contract]: string };
  ...
  const ap = contracts["multisig/ap"]
```

Reply:
> A middleground probably would be to retain the static list but it's value is directly from json

This can be done, would allow us to at least not have to constantly copy/paste new contract addresses into the webapp, but have it done semi-automatically (still have to approve a PR).